tracee/printer.go

```diff
@@ -112,11 +86,18 @@ func newEvent(ctx context, argsNames []string, args []interface{}) (Event, error
 		EventName:           EventsIDToEvent[int32(ctx.EventID)].Name,
 		ArgsNum:             int(ctx.Argnum),
 		ReturnValue:         int(ctx.Retval),
-		Args:                make([]Argument, 0, len(args)),
+		Args:                make([]external.Argument, 0, len(args)),
+	}
+	params, ok := EventsIDToParams[ctx.EventID]
+	if !ok {
+		return e, fmt.Errorf("event %s(%d) is missing from params registry", e.EventName, e.EventID)
 	}
 	for i, arg := range args {
-		e.Args = append(e.Args, Argument{
-			Name:  argsNames[i],
+		e.Args = append(e.Args, external.Argument{
+			ArgMeta: external.ArgMeta{
+				Name: params[i].Name,
```

Comment:
You can't assume the arg name to be found according to its index.
There are cases where the argument number changes for the same event.
for example: envp in execve event is only shown if the user chooses to do that.
Another example: bpf code failed to read some argument (e.g. string), thus this argument will not be sent at first place